elasticsearch-details/groups-rule-mapping.md

```diff
@@ -0,0 +1,76 @@
+# External to local groups mapping 
+
+`Groups` rule matches request which user is assigned to at least one of its groups. Groups are being assigned to a username (or username pattern) in `users` section of ROR's configuration. Each element of the section requires also an authentication method (like eg. `auth_key` or `ldap_authentication`). But we can also use an authorization rule in this section beside the mentioned authentication rule (or one rule which provides both - the authentication and authorization altogether). Thanks to this, we gain a possibility to map external groups to local ones. Let's see it on an example.
```

Comment:
*ok, this needs rewording...*
 
The `groups` accepts a list of group names. This rule will match requests in which the resolved username belongs at least to one of the listed groups. The association between usernames and groups is explicitly declared in the `users` section of the ACL. This is a list of usernames (full wildcard patterns are supported). 
Each entry in the association rule requires:
* An authenticaton rule:  (I.e. auth_key_* for local credentials, or ldap_authentication, external_authentication,aa etc) 
* A list of groups within the ones precendently inserted in the groups rules of the ACL blocks
* Optional: a remote authorization rule (ldap_authorization, groups_provider_authorization, etc.)

When the groups rule and the authorization rule are used together, we obtain "group mapping". That is: we are effectively mapping remote groups to local groups.

Note: the rule `ldap_auth` is the composition of `ldap_authentication` and `ldap_authorization`. So it can be used as a shortcut for both.